File: ABC-229/B.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  string A, B;
  cin >> A >> B;

  int n = min(A.length(), B.length());
  reverse(A.begin(), A.end());
  reverse(B.begin(), B.end());

  for (int i = 0; i < n; ++i) {
    int a, b;
    a = int(A[i] - '0');
    b = int(B[i] - '0');
    if (a + b >= 10) {
      cout << "Hard" << endl;
      return 0;
    }
  }
  cout << "Easy" << endl;
  return 0;
}

File: ABC-229/B.py
a, b = input().split()

a = a[::-1]
b = b[::-1]

n = min(len(a), len(b))

for i in range(n):
    _a, _b = int(a[i]), int(b[i])
    if _a + _b >= 10:
        print("Hard")
        exit()

print("Easy")
